File: TUTORIAL_5/WORKSPACE/pyTelescope/src/acstut/TelescopeImpl.py
import acstutorial
import acstutorial__POA
import acstut
import CUSTOMErr
import CUSTOMErrImpl
  
from Acspy.Servants.ACSComponent import ACSComponent
from Acspy.Servants.ContainerServices import ContainerServices
from Acspy.Servants.ComponentLifecycle import ComponentLifecycle
  
class TelescopeImpl(acstutorial__POA.Telescope, ACSComponent, ContainerServices, ComponentLifecycle):
    def __init__(self):
        ACSComponent.__init__(self)
        ContainerServices.__init__(self)
        self._logger = self.getLogger()
        self.x = 0
        self.y = 0

    def printHello(self):
        print("Just printing 'Hello World!'")
        return "Hello World!"

    def moveTo(self, x, y):

        if x < 0 or y < 0:
            raise CUSTOMErrImpl.PositionOutOfLimitsExImpl()
        self.x = x
        self.y = y
    
    def getCurrentPosition(self):
        print("Telescope current position in x:" + str(self.x)+ "y: " + str(self.y))
        return "Telescope in: x: " + str(self.x) + " y: " + str(self.y)

File: TUTORIAL_2/WORKSPACE/jTelescope/src/acsws/acstutorial/TelescopeImpl/TelescopeImpl.java
package acsws.acstutorial.TelescopeImpl;

import alma.acs.component.ComponentImplBase;
import acsws.acstutorial.TelescopeOperations;

public class TelescopeImpl extends ComponentImplBase implements TelescopeOperations
{
    private float x;
    private float y;

    public TelescopeImpl()
    {
        this.x = 0;
        this.y = 0;
    }

    public void moveTo(float x, float y)
    {
        this.x = x;
        this.y = y;
    }

    public String getCurrentPosition()
    {   
        String position = "Telescope in x:" + this.x + " y:" + this.y;
        System.out.println(position);
        return position;
    }
}